### ISSUE TITLE
Invalid mapping since upgrade 3.4.14 to 3.4.18

### ISSUE THREAD

**Issue by MWMathieu** [symfony/symfony#29289]
**Symfony version(s) affected**: 3.4.18

**Description**  
Since i upgrade only symfony/symfony 3.4.14 to 3.4.18, command doctrine:schema:validate return some errors. When i rollback in 3.4.14 mapping is valid.

> [Mapping]  FAIL - The entity-class SmartEventBundle\Model\News\File mapping is invalid:
> * The association SmartEventBundle\Model\News\File#news refers to the inverse side field SmartEventBundle\Model\News\News#files which does not exist.

**How to reproduce** 

- MappingListener :  

```php
if ($className === $this->newsClass) {
            $this->setConcrete($metadata, $this->newsTable);
           $metadata->mapOneToMany([
               'targetEntity' => $this->newsFileClass,
               'fieldName'    => 'files',
               'mappedBy'     => 'news',
               'cascade'      => ['persist', 'remove'],
           ]);
}

if ($className === $this->newsFileClass) {
            $this->setConcrete($metadata, $this->newsFileTable);
            $metadata->mapManyToOne([
                'targetEntity' => $this->newsClass,
                'fieldName'    => 'news',
                'inversedBy'   => 'files',
            ]);
}
```

- News model : 

```php
    /**
     * @var ArrayCollection|FileInterface[]
     */
    protected $files;
```

- File model : 

```php
    /**
     * @var NewsInterface
     */
    protected $news;
```

**Additional context**  
Dynamic mapping with PHP in doctrine listener for event : loadClassMetadata

**Comment by xabbuh** [comment]
Are you sure that only Symfony packages were updated? Maybe you also updated some Doctrine related packages which led to the different behaviour. Where does this `MappingListener` class live that you refer to? Could you create a small example application that allows to reproduce your issue?

**Comment by nicolas-grekas** [comment]
Here are some: https://github.com/symfony/doctrine-bridge/compare/v3.4.14...v3.4.18

**Comment by MWMathieu** [comment]
Nothing about mapping and validation in this change. Just doctrine/common stable dependency removed.

**Comment by ro0NL** [comment]
Can you confirm the listener is actually called. Might be a DI issue as well :thinking:

**Comment by MWMathieu** [comment]
Yes i confirm Listener is called other field mappings is ok.

**Comment by ro0NL** [comment]
Then you can also debug where its execution differs between SF versions...no? if that's the only updated dependency.

**Comment by MWMathieu** [comment]
After searching I realized that some of my models were no longer in the list of $eventArgs->getClassMetadata();

In the same listener some models registered in classMetadata and other not listed.

**Comment by MWMathieu** [comment]
But when i run php bin/console doctrine:mapping:info all my entities are listed.

**Comment by MWMathieu** [comment]
Problem fix in 3.4.19. I dont't know what has been fixed @xabbuh @nicolas-grekas

**Comment by nicolas-grekas** [comment]
Great news, must be #29247 or #29104
